src/langchain_google_spanner/vector_store.py

```diff
@@ -985,9 +1286,25 @@ def similarity_search_by_vector(
         Returns:
             List[Document]: List of documents most similar to the query.
         """
-        documents = self.similarity_search_with_score_by_vector(
-            embedding=embedding, k=k, pre_filter=pre_filter
-        )
+        documents: List[Tuple[Document, float]] = []
+        if self.__using_ANN:
```

Comment:
Please remove it from here  as you will already this check in `max_marginal_relevance_search_with_score_by_vector` and `similarity_search_with_score_by_vector`

Reply:
@gauravpurohit06 could we please do that in a follow up PR? I plan on mailing it, just that this one has become too huge